src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Stratis.Bitcoin.IntegrationTests.Common;
+using Stratis.Bitcoin.IntegrationTests.Common.EnvironmentMockUpHelpers;
+using Stratis.Bitcoin.Networks;
+using Stratis.Bitcoin.Primitives;
+using Xunit;
+
+namespace Stratis.Bitcoin.IntegrationTests
+{
+    public sealed class ConsensusManagerInterceptorTests
+    {
+        /// <summary>
+        /// In this scenario we test what happens when a node disconnected during a rewind and before it
+        /// tries to connect to another chain with longer chain work.
+        /// </summary>
+        [Fact]
+        public void ReorgChain_AfterInitialRewind_Miner_Disconnects()
+        {
+            using (var builder = NodeBuilder.Create(this))
+            {
+                var minerA = builder.CreateStratisPosNode(new StratisRegTest()).NotInIBD().WithWallet();
+                var minerB = builder.CreateStratisPosNode(new StratisRegTest()).NotInIBD().NoValidation().WithWallet();
+                var minerC = builder.CreateStratisPosNode(new StratisRegTest()).NotInIBD().WithWallet();
+
+                // Configure the interceptor to disconnect a node after a certain block has been disconnected (rewound).
+                bool interceptor(ChainedHeaderBlock chainedHeaderBlock)
+                {
+                    if (chainedHeaderBlock.ChainedHeader.Previous.Height == 5)
+                    {
+                        TestHelper.Disconnect(minerA, minerC);
+                        return true;
+                    }
+
+                    return false;
+                }
+                minerA.BlockDisconnectInterceptor(interceptor);
```

Comment:
Would be cleaner if you did 
```
 var minerA = builder.CreateStratisPosNode(new StratisRegTest()).NotInIBD().WithWallet().BlockDisconnectInterceptor(interceptor).Start();
```

Or even just call it `Intercept(`

Reply:
In this case I need to first create the node that I wish to disconnect from (minerA) so I first have to set up the nodes and then inject the interceptor function.

Also, I want to be specific on what the interceptor is intercepting, so thought it better to call `BlockDisconnectInterceptor`?

Thoughts?

Reply:
ah yeah its going to be tricky so like this is good